package.json

```diff
@@ -24,8 +24,10 @@
 		"README.md"
 	],
 	"scripts": {
-		"build": "tsc",
```

Comment:
Removing `tsc` altogether from `package.json` means there's nothing running type checking anymore! 😱 

In previous repos I've used `compile: "tsc"`. How about...
* Enabling `noEmit: true` in the `tsconfig.json`
* Adding a `compile.yml` workflow
* Removing `build:clean` (since now only these build scripts write to `lib/`

Reply:
`tsup` runs type checking when the `--dts` flag is used ([as documented here](https://tsup.egoist.dev/#tree-shaking:~:text=Additionally%2C%20if%20you%20want%20type%20checking%20at%20build%20time%2C%20you%20can%20enable%20%2D%2Ddts%2C%20which%20will%20run%20a%20real%20TypeScript%20compiler%20to%20generate%20declaration%20file%20so%20you%20get%20type%20checking%20as%20well.)).

We can add an explicit type checking script though to make sure type checking is happening.

Reply:
I added an explicit type check script and workflow. I just call them "typecheck".

Reply:
Ah! Sorry you're totally right about `--dts` doing that, I'd misread/mis-skimmed the docs.

But, I'd still want to run `tsc` directly. `tsup`'s build output isn't particularly nice for type errors. Adding `export const value: string = 7;` to a file:

```plaintext
CJS ⚡️ Build success in 10ms
src/comments.ts(8,14): error TS2322: Type 'number' is not assignable to type 'string'.

DTS Build failed
RollupError: Failed to compile. Check the logs above.
    at error (/Users/josh/repos/ts-api-utils/node_modules/.pnpm/rollup@3.9.1/node_modules/rollup/dist/shared/rollup.js:210:30)
    at Object.error (/Users/josh/repos/ts-api-utils/node_modules/.pnpm/rollup@3.9.1/node_modules/rollup/dist/shared/rollup.js:23975:20)
    at Object.error (/Users/josh/repos/ts-api-utils/node_modules/.pnpm/rollup@3.9.1/node_modules/rollup/dist/shared/rollup.js:23220:42)
    at Object.transform (/Users/josh/repos/ts-api-utils/node_modules/.pnpm/tsup@6.5.0_typescript@4.9.5/node_modules/tsup/dist/rollup.js:7287:20)
    at /Users/josh/repos/ts-api-utils/node_modules/.pnpm/rollup@3.9.1/node_modules/rollup/dist/shared/rollup.js:24176:40
```

Especially for newcomers, this isn't idea.